src/components/PriceLine.tsx

```diff
@@ -54,32 +58,72 @@ const PriceLine = ({
                 className={classNames('chart-line', 'horizontal', className || '', {
                     draggable,
                     dragging: isDragging,
                 })}
                 style={{
                     color: foregroundColor,
-                    backgroundImage:
-                        lineStyle && lineStyle !== 'solid' ? '' : `linear-gradient(to left, ${color} 90%, ${color}00`,
                 }}
             >
                 {showBarrierDragLine && (
-                    <div className='drag-line' style={{ borderTop: `${lineStyle} ${color} 1px` }} />
+                    <div
+                        className={classNames('drag-line', { 'drag-line--zero': isOverlappingWithPriceLine })}
+                        style={{
+                            borderTopColor: color,
+                            borderTopStyle: lineStyle as React.CSSProperties['borderTopStyle'],
+                            width: `calc(100% - ${width}px + ${!isMobile ? overlappedBarrierWidth : 0}px)`,
+                        }}
+                    />
                 )}
                 <div className='draggable-area' />
-                <div className='drag-price' style={{ backgroundColor: color, width, opacity }}>
-                    <div className='price'>{priceDisplay}</div>
-                    {offScreen && offScreenDirection && (
-                        <PriceLineArrow offScreenDirection={offScreenDirection} color={color} />
+                <div className='draggable-area-wrapper'>
+                    <div
+                        className={'drag-price'}
+                        style={{
+                            backgroundColor: color,
+                            width: isOverlappingWithPriceLine ? overlappedBarrierWidth : width,
+                            opacity,
+                            right: (isOverlappingWithPriceLine ? width - overlappedBarrierWidth : 0) + (isMobile ? 20 : 4),
+                        }}
+                    >
+                        <HamburgerDragIcon />
+                        <div
+                            className={classNames('price', { 'price--zero': isOverlappingWithPriceLine })}
+                            style={{
+                                color: isOverlappingWithPriceLine ? color : '',
+                                right: isOverlappingWithPriceLine
+                                    ? overlappedBarrierWidth + priceDisplay.length * 8 - (!draggable ? 16 : 0)
+                                    : 0,
+                            }}
+                        >
+                            {priceDisplay}
+                        </div>
+                        <div />
+                        {offScreen && offScreenDirection && (
+                            <PriceLineArrow offScreenDirection={offScreenDirection} color={color} />
+                        )}
+                    </div>
+                    {isOverlappingWithPriceLine && (
+                        <div>
```

Comment:
Sorry if it's irrelevant, do we need extra empty `<div>` tag here?

Reply:
I removed it, looks like it's not in use anymore